player/public/js/baseLauncher.js

```diff
@@ -128,8 +130,12 @@ org.ekstep.contentrenderer.baseLauncher = Class.extend({
 		if (this.sleepMode) return
 		if (TelemetryService.instance && TelemetryService.instance.telemetryStartActive()) {
 			var telemetryEndData = {}
+			var endpageSeen = [{ "endpageseen" : this.isEndPageSeen }]
+			var contentProgress = [{ "progress" : this.contentProgress() }]
+			var contentSummary = (_.isFunction(this.contentPlaySummary)) ? this.contentPlaySummary() : ''
```

Comment:
Define the interface methods for each of the common summary properties, similar to contentProgress. Each launcher has to override/implement these methods.

Then addition summary specific to contentType will be given by another interface method additionalContentSummary().

Reply:
Base launcher should create common summary event. So that for any content, these fields will exist.